pennylane/resources/first_quantization.py

```diff
@@ -0,0 +1,192 @@
+    vectors, :math:`G_0`.
+
+    For :math:`\lambda_{\nu}` defined in Eq. (25) as
+
+    .. math::
+
+        \lambda_{\nu} = \sum_{\nu \in G_0} \frac{1}{\left \| \nu \right \|^2},
+
+    we use
+
+    .. math::
+
+        \lambda_{\nu} = 4\pi \left ( \frac{\sqrt{3}}{2} N^{1/3} - 1 \right) + 3 - \frac{3}{N^{1/3}}
+        + 3 \int_{x=1}^{N^{1/3}} \int_{y=1}^{N^{1/3}} \frac{1}{x^2 + y^2} dydx.
+
+    We also need to compute :math:`\lambda^{\alpha}_{\nu}` defined in Eq. (123)
+
+    .. math::
+
+        \lambda^{\alpha}_{\nu} = \alpha \sum_{\nu \in G_0} \frac{\left \lceil M(2^{\mu - 2}) / \left
+        \| \nu \right \|^2 \right \rceil}{M2^{2\mu - 4}},
+
+    which we compute here, following Eq. (113) for :math:`\alpha = 1`, as
+
+    .. math::
+
+        \lambda^{1}_{\nu} = \lambda_{\nu} + \frac{4}{2^{n_m}} (7 \times 2^{n_p + 1} + 9 n_p -
+        11 - 3 \times 2^{-n_p}),
+
+    where :math:`M = 2^{n_m}` and :math:`n_m` is defined in Eq. (132). Finally, for :math:`p_{\nu}`
+    defined in Eq. (128)
+
+    .. math::
+
+        p_{\nu} = \sum_{\mu = 2}^{n_p + 1} \sum_{\nu \in B_{\mu}} \frac{\left \lceil M(2^{\mu-2} /
+        \left \| \nu \right \|)^2 \right \rceil}{M 2^{2\mu} 2^{n_{\mu} + 1}},
+
+    we use the upper bound from Eq. (29) in [`arXiv:1807.09802v2 <https://arxiv.org/abs/1807.09802v2>`_]
+    which gives :math:`p_{\nu} = 0.2398`.
+
+    Args:
+        eta (int): number of electrons
+        n (int): number of basis states
+        omega (float): unit cell volume
+        error (float): target error in the algorithm
+        br (int): number of bits for ancilla qubit rotation
+        charge (int): total electric charge of the system
+
+    Returns:
+        float: 1-norm of a first-quantized Hamiltonian in the plane-wave basis
+
+    **Example**
+
+    >>> eta = 156
+    >>> n = 10000
+    >>> omega = 1145.166
+    >>> error = 0.001
+    >>> norm_fq(eta, n, omega, error)
+    1254385.059691027
+    """
+    if n <= 0 or not isinstance(n, int):
+        raise ValueError("The number of planewaves must be a positive integer.")
+
+    if eta <= 0 or not isinstance(n, int):
+        raise ValueError("The number of electrons must be a positive integer.")
+
+    if omega <= 0:
+        raise ValueError("The unit cell volume must be a positive number.")
+
+    if error <= 0.0:
+        raise ValueError("The target error must be greater than zero.")
+
+    if br <= 0 or not isinstance(br, int):
+        raise ValueError("br must be a positive integer.")
+
+    if not isinstance(charge, int):
+        raise ValueError("system charge must be an integer.")
+
+    l_z = eta + charge
+
+    # target error in the qubitization of U+V which we set to be 0.01 of the algorithm error
+    error_uv = 0.01 * error
+
+    # taken from Eq. (22) of PRX Quantum 2, 040332 (2021)
+    n_p = int(np.ceil(np.log2(n ** (1 / 3) + 1)))
+
+    n0 = n ** (1 / 3)
+    lambda_nu = (
```

Comment:
We should give credit to T-Fermion here since we are using their approximation of lambda_nu.